### PR TITLE
Error message for an incorrectly entered username

### PR DESCRIPTION
**Pull request by treeshateorcs**
# Problem
https://tickets.metabrainz.org/browse/LB-1130

# Solution

# Action

### CODE REVIEW

**Review by amCap1712 — CHANGES_REQUESTED** [review]
Hi! Thanks for the PR but it seems incorrect in the current form. I suggest to make the changes as per the other comment and try and test locally if works as expected. Once that is done, I can help you add a test for this as well. Feel free to ask any other questions you may have.

File: listenbrainz/webserver/static/js/src/lastfm/LastFMImporter.tsx [hunk 1]
```diff
@@ -122,17 +122,24 @@ export default class LastFmImporter extends React.Component<
...
       const data = await response.json();
+      if ("message" in data && data.message === "User not found") {
+        const wrong_username = true;
+      }
       if ("playcount" in data.user) {
         return Number(data.user.playcount);
       }
       return -1;
-    } catch {
+    } catch (wrong_username) {
```
[the remark this answers is not in the document]
Comment (treeshateorcs) [reply listenbrainz/webserver/static/js/src/lastfm/LastFMImporter.tsx]: oh no, how could i not think of placing wrong_username in a different scope! thank you! all done now

**Comment by amCap1712** [comment]
The latest changes look good to me. Thanks. 

I think we should also add a test for this to avoid regressions. Are you familiar with how to write a jest test?

It should test that the the correct error message is indeed displayed for the invalid username error. It will probably be similar to this test: https://github.com/metabrainz/listenbrainz-server/blob/d8a287962dd5e1523ee9d60ffa55c942a558ad38/listenbrainz/webserver/static/js/tests/lastfm/LastFMImporter.test.tsx#L120-L132

I can help you write one or add one myself if you prefer.

**Comment by treeshateorcs** [comment]
okay, so i'm about to add a test. so far i have this, but i'm not sure about some things

```javascript
  it("should return user not found", async () => {
    instance.setState({ lastfmUsername: "nonexistentusernamedonttryathome" });

    await expect(instance.getTotalNumberOfScrobbles()).rejects.toThrowError(); 
    expect(instance.state.msg?.props.children).toMatch(  // not sure what this has to be
       "User not found" 
     ); 
  })
```

**Comment by treeshateorcs** [comment]
or maybe this is better?

```javascript
  it("should return user not found", async () => {
    instance.setState({ lastfmUsername: "nonexistentusernamedonttryathome" });

    try {
      await instance.getTotalNumberOfScrobbles();
    } catch (err) {
      expect(err).toEqual(
        new Error(
          `User not found`
        )
      );
    }
  })
```


i've never seriously worked with javascript tbh

**Review by amCap1712 — APPROVED** [review]
Thanks for adding the test. It was missing a mock to generate the error message so I updated it.

**Pull request merged**